src/Toolkit/Toolkit/UI/Controls/Legend/Legend.cs

```diff
@@ -14,25 +14,128 @@
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+#elif __IOS__
+using Control = UIKit.UIView;
+#elif __ANDROID__
+using Control = Android.Views.ViewGroup;
+#else
+using System.Windows;
+using System.Windows.Controls;
+#endif
+
 namespace Esri.ArcGISRuntime.Toolkit.UI.Controls
 {
     /// <summary>
     /// The Legend control is used to display symbology and description for a set of <see cref="Layer"/>s
     /// in a <see cref="Map"/> or <see cref="Scene"/> contained in a <see cref="GeoView"/>.
     /// </summary>
-    public partial class Legend : LayerList
+    public partial class Legend
     {
+        private readonly LegendDataSource _datasource = new LegendDataSource(null);
+
 #if !__ANDROID__
         /// <summary>
         /// Initializes a new instance of the <see cref="Legend"/> class.
         /// </summary>
         public Legend()
-            : base()
         {
+            Initialize();
         }
 #endif
+
+#if XAMARIN
+        private GeoView _geoView;
+#endif
+
+        /// <summary>
+        /// Gets or sets the geoview that contain the layers whose symbology and description will be displayed.
+        /// </summary>
+        /// <seealso cref="MapView"/>
+        /// <seealso cref="SceneView"/>
+        public GeoView GeoView
+        {
+#if XAMARIN
+            get => _geoView;
+            set
+            {
+                if (_geoView != value)
+                {
+                    var oldView = _geoView;
+                    _geoView = value;
+                    _datasource.SetGeoView(value);
+                }
+            }
+#else
+            get { return (GeoView)GetValue(GeoViewProperty); }
+            set { SetValue(GeoViewProperty, value); }
+#endif
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the scale of <see cref="GeoView"/> and any scale ranges on the <see cref="Layer"/>s
+        /// are used to determine when legend for layer is displayed.
+        /// </summary>
+        /// <remarks>
+        /// If <c>true</c>, legend for layer is displayed only when layer is in visible scale range;
+        /// otherwise, <c>false</c>, legend for layer is displayed regardless of its scale range.
+        /// </remarks>
+        public bool FilterByVisibleScaleRange
+        {
+#if XAMARIN
+            get => _datasource.FilterByVisibleScaleRange;
+            set => _datasource.FilterByVisibleScaleRange = value;
+#else
+            get { return (bool)GetValue(FilterByVisibleScaleRangeProperty); }
+            set { SetValue(FilterByVisibleScaleRangeProperty, value); }
+#endif
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the visibility of a <see cref="Layer"/>
+        /// is used to determine when the legend for the layer is displayed.
+        /// </summary>
+        /// <remarks>
+        /// If <c>true</c>, legend for layer is displayed only when the layer's <see cref="Layer.IsVisible"/> property is true.
```

Comment:
Does this apply to sublayers (`ILayerContent.IsVisible`) too? Should we doc that?